svc/svc.go

```diff
@@ -107,21 +108,40 @@ func NewStandardHandler(opts HandlerOpts) http.Handler {
 	return middleware.BackgroundContext(h)
 }
 
-// RunServer handles the biolerplate of starting an http server and handling
-// signals gracefully.
-func RunServer(h http.Handler, port string, writeTimeout time.Duration) {
-	// Add timeouts to the server
-	srv := &http.Server{
-		WriteTimeout: writeTimeout * time.Second,
-		Addr:         ":" + port,
-		Handler:      h,
+// BuildServerOpt allows users to customize the http.Server used by RunServer.
+type BuildServerOpt func(*http.Server)
+
+// ServerDefaults specifies default server options to use for RunServer.
+var ServerDefaults = func(srv *http.Server) {
+	srv.Addr = ":8080"
+	srv.WriteTimeout = 5 * time.Second
+	srv.ReadHeaderTimeout = 5 * time.Second
+	srv.IdleTimeout = 120 * time.Second
+}
+
+// WithPort sets the port for the server to run on.
+func WithPort(port string) BuildServerOpt {
+	return func(srv *http.Server) {
+		srv.Addr = ":" + port
 	}
+}
 
-	fmt.Printf("HTTP server listening on port %s\n", port)
-	runServer(srv)
+// BuildServer offers some convenience and good defaults for creating an http.Server
+func BuildServer(h http.Handler, opts ...BuildServerOpt) *http.Server {
```

Comment:
how about calling it `NewServer`? it would be more consistent with other factory like functions in Go, I think